### PR TITLE
Add godoc for missing Pipeline fields

### PR DESCRIPTION
**Pull request by pmorie**
# Changes

Adds godoc for some fields in the `Pipeline` API that were missing them.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

**Comment by vdemeester** [comment]
/ok-to-test
:heart:

File: pkg/apis/pipeline/v1alpha1/pipeline_types.go [hunk 1]
```diff
@@ -94,12 +107,17 @@ type PipelineTaskParam struct {
 	Value string `json:"value"`
 }
 
-// PipelineParam defines arbitrary parameters needed by a Pipeline beyond typed inputs
+// PipelineParam defines an arbitrar parameter needed by a Pipeline beyond typed inputs
```
Comment (sbwsg) [inline pkg/apis/pipeline/v1alpha1/pipeline_types.go]: arbitrar -> arbitrary

Comment (pmorie) [reply pkg/apis/pipeline/v1alpha1/pipeline_types.go]: whoops

**Pull request merged**